app/components/post_list/post_list.android.js

```diff
@@ -0,0 +1,131 @@
+        if (this.props.channelId !== nextProps.channelId) {
+            this.contentOffsetY = 0;
+            this.contentHeight = 0;
+        }
+
+        if (nextProps.postIds !== this.props.postIds && this.mounted) {
+            this.setState({
+                dataSource: new DataSource(nextProps.postIds, this.keyExtractor),
+            });
+        }
+    }
+
+    handleLayout = (layoutHeight, contentWidth, contentHeight) => {
+        this.contentHeight = contentHeight;
+
+        if (layoutHeight && contentHeight < layoutHeight) {
+            // We still have less than 1 screen of posts loaded with more to get, so load more
+            this.props.onLoadMoreUp();
+        }
+    };
+
+    handleScroll = (event) => {
+        const {layoutMeasurement, contentOffset, contentSize} = event.nativeEvent;
+
+        if (!this.postListHeight) {
+            this.handleLayout(layoutMeasurement.height, contentSize.width, contentSize.height);
+        }
+
+        this.postListHeight = layoutMeasurement.height;
+
+        if (contentOffset.y >= 0) {
+            const definedHeight = Math.round(this.postListHeight) * SCROLL_UP_MULTIPLIER;
+            const pageOffsetY = contentOffset.y;
+            this.contentHeight = contentSize.height;
+            const direction = (this.contentOffsetY < pageOffsetY) ?
+                ListTypes.VISIBILITY_SCROLL_DOWN :
+                ListTypes.VISIBILITY_SCROLL_UP;
+            this.contentOffsetY = contentOffset.y;
+
+            switch (direction) {
+            case ListTypes.VISIBILITY_SCROLL_DOWN:
+                if ((Math.round(this.contentHeight - pageOffsetY) < definedHeight)) {
+                    this.props.onLoadMoreDown();
+                }
+                break;
+            case ListTypes.VISIBILITY_SCROLL_UP:
+                if (Math.round(pageOffsetY) < definedHeight) {
+                    this.props.onLoadMoreUp();
+                }
+                break;
+            }
+        }
+    };
+
+    render() {
+        const {
+            channelId,
+            initialIndex,
+            postIds,
+        } = this.props;
+
+        const otherProps = {};
+        if (postIds.length) {
+            otherProps.ListFooterComponent = this.props.renderFooter();
+        } else {
+            otherProps.ListEmptyComponent = this.props.renderFooter();
+        }
+
+        const hasPostsKey = postIds.length ? 'true' : 'false';
+        return (
+            <RecyclerRefreshControl
+                key={`recyclerFor-${channelId}-${hasPostsKey}`}
+                colors={['red', 'blue']}
+                direction='bottom'
+                enabled={Boolean(channelId)}
+                onRefresh={this.handleRefresh}
+                refreshing={this.state.refreshing}
+                style={styles.flex}
+            >
+                <RecyclerViewList
+                    key={`${channelId}-${hasPostsKey}`}
+                    ref={'list'}
+                    style={{flex: 1}}
```

Comment:
Could use the flex in the styles object below.

Reply:
Also noticed the padding on the iOS component, does that need to be added here as well to make sure the most recent post has the same spacing from the bottom?

Reply:
cool will do, thanks